core/src/main/java/org/apache/iceberg/RewriteTablePathUtil.java

```diff
@@ -51,6 +50,8 @@
 public class RewriteTablePathUtil {
 
   private static final Logger LOG = LoggerFactory.getLogger(RewriteTablePathUtil.class);
+  // iceberg client file separator, independent of that of the underlying file system
```

Comment:
I'm not sure this comment is helpful? We don't have an actual defined "Iceberg Client File Separator". This is a POSIX path separator that makes sense to use because AWS, GCP, AZURE, HDFS etc ... all use it. Comments shouldn't beg the question, they should ask and answer it if needed. 

In this case something like

"Use the POSIX separator instead of File.separator because File.separator is dependent on the client environment and not the target filesystem. POSIX is compatible with S3, GCS, HDFS, etc ..."

So we describe what we are doing and why we are doing that instead of something else.